client/src/main/java/tds/assessment/Item.java

```diff
@@ -137,10 +141,47 @@ public void setItemProperties(List<ItemProperty> itemProperties) {
     }
 
     /**
+     * @return The full path and file name of the item's XML file
+     */
+    public String getItemFilePath() {
+        return itemFilePath;
+    }
+
+    public void setItemFilePath(String itemFilePath) {
+        this.itemFilePath = itemFilePath;
+    }
+
+    /**
+     * @return The full path and file name of the item's stimulus file (if there is one)
+     */
+    public String getStimulusFilePath() {
+        return stimulusFilePath;
+    }
+
+    public void setStimulusFilePath(String stimulusFilePath) {
+        if (stimulusFilePath == null) {
```

Comment:
Technically, the `tblitem.filepath` column is nullable, so it probably should get the same treatment.  As it turns out, all items in `itembank.tbltem` have a `filepath` value.  Same for `itembank.tblstimulus`.  However, there are Items that do not have a "companion" stimulus, which is why the query does a `LEFT JOIN` when fetching stimulus information.  Based on the data we have available, the assumption is that an Item will always have an `itemFilePath` but not necessarily a `stimulusFilePath`.

But the nullable `itembank.tblitem.filepath` trumps all that, so I'll add a `null` check.